base/distributed/cluster.jl

```diff
@@ -210,13 +210,27 @@ end
 # The master process uses this to connect to the worker and subsequently
 # setup a all-to-all network.
 function read_worker_host_port(io::IO)
-    while true
-        conninfo = readline(io)
+    t0 = time()
+    timeout = worker_timeout()
+
+    ntries = 1000
```

Comment:
that should at least be put in a comment to explain what the code is doing and why 1000 is a reasonable value to use